### ISSUE TITLE
Istio 1.24 RC - Pilot References Still Present in injection-template.yaml

### ISSUE THREAD

**Issue by CWoodall2100** [istio/istio#53791]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

While testing with Istio 1.24 RC 0, I noticed that there are still references to pilot in [injection-template.yaml](https://github.com/istio/istio/blob/2def51733077f350382dcf8f68ae7d1a3d3c6f99/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml) in the istiod helm chart.

Since the pilot section of values.yaml has been refactored (removed) for 1.24, we should remove these last few references. Should be `.Values.cni` instead of `.Values.pilot.cni`
i.e. https://github.com/istio/istio/blob/2def51733077f350382dcf8f68ae7d1a3d3c6f99/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L48

### Version

```Text
N/A
```

### Additional Information

N/A

**Comment by hzxuzhonghu** [comment]
cc @bleggett

**Comment by bleggett** [comment]
Yeah that's what https://github.com/istio/istio/pull/53611 is for.

cc @nshankar13 do you want to keep that PR open and fixup or do you want me to do it?

**Comment by nshankar13** [comment]
Will take another stab at this.
